browser/android/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/android/brave_cosmetic_resources_tab_helper.h"
+
+#include <string>
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+
+
+namespace {
+bool IsAdBlockEnabled(content::WebContents* contents, const GURL& url) {
+  Profile* profile = static_cast<Profile*>(contents->GetBrowserContext());
+
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+  brave_shields::ControlType control_type =
+      brave_shields::GetAdControlType(map, contents->GetLastCommittedURL());
+  if (brave_shields::GetBraveShieldsEnabled(map, url) &&
+      control_type != brave_shields::ALLOW) {
+    content::RenderFrameHost::AllowInjectingJavaScript();
+
+    return true;
+  }
+
+  return false;
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+		const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources = g_brave_browser_process->
+      ad_block_service()->UrlCosmeticResources(url);
+
+  if (!resources || !resources->is_dict()) {
+    return result_list;
+  }
+
+  base::Optional<base::Value> regional_resources = g_brave_browser_process->
+      ad_block_regional_service_manager()->UrlCosmeticResources(url);
+
+  if (regional_resources && regional_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*regional_resources), &*resources, /*force_hide=*/false);
+  }
+
+  base::Optional<base::Value> custom_resources = g_brave_browser_process->
+      ad_block_custom_filters_service()->UrlCosmeticResources(url);
+
+  if (custom_resources && custom_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*custom_resources), &*resources, /*force_hide=*/true);
+  }
+
+  result_list->Append(std::move(*resources));
+
+  return result_list;
+}
```

Comment:
We recently moved the regional and custom filters services into the main adblock service; the majority of this function ought to be moved into `AdBlockService::UrlCosmeticResources` so we can share the same logic on desktop. I can handle this in a follow-up if you'd prefer.

Reply:
yeah it would be good if you can handle, as I'm not certain of what should be done exactly. Should it be just a one call to something instead of regional and custom calls?

Reply:
correct - don't worry about it, I'll assign an issue for myself